src/encoded/audit/genetic_modification.py

```diff
@@ -0,0 +1,34 @@
+from snovault import (
+    AuditFailure,
+    audit_checker,
+)
+
+
+@audit_checker('GeneticModification', frame='object')
+def audit_genetic_modification_reagents(value, system):
+    '''
+    Genetic modifications with missing reagents will be flagged by this audit.
+    Modifications with acceptable substitutes for reagents will not be flagged,
+    depending on their method. See the method dependecy in file.json for more
+    detail.
+    '''
+    missing_reagents = False
+
+    if not value.get('reagents'):
```

Comment:
See if you can combine all of the conditionals in 'any'
```
if not value.get('reagents'):
    method = value['method']
    if any([
            method == 'CRISPR' and not value.get('guide_rna_sequences'),
            method == 'RNAi' and not value.get('rnai_sequences'),
            method == 'TALEN',
            not value.get('documents'),
        ]):
        detail = 'Genetic modification {} of method {} is missing reagents'.format(
            value['@id'],
            method,
        )
        yield AuditFailure('missing genetic modification reagents', detail, level='ERROR')
```
Another python built in is 'all'.  Which would require all the conditions to be true.  'any' will return true after finding the first true condition.  Don't trust the code above, I didn't test it.

Reply:
It would be nicer if I could combine the conditionals into an 'any', but checking for 'documents' should only be performed for methods other than 'CRISPR', 'TALEN', and 'RNAi'. The logic above would return True if the 'method' is CRISPR and 'value' had 'guide_rna_sequences' but not 'documents', which is not the desired behavior.

Reply:
nice